File: Intent/app/src/main/java/com/example/intent/MainActivity2.java
package com.example.intent;

import android.content.Intent;
import android.os.Bundle;

import com.google.android.material.floatingactionbutton.FloatingActionButton;
import com.google.android.material.snackbar.Snackbar;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;

import android.view.View;
import android.widget.Button;

public class MainActivity2 extends AppCompatActivity {


    Button button;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main2);
        button = (Button)findViewById(R.id.button2);

        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

//************ Intent ************\\
// Explicit
//                Intent intent = new Intent(MainActivity2.this,MainActivity.class);
//                startActivity(intent);

// Implicit
                Intent i = getPackageManager().getLaunchIntentForPackage("com.example.popupmenu");
                startActivity(i);
            }
        });
    }
}

File: kotlinAssignment/app/src/main/java/com/example/kotlinassignment/MainActivity2.kt
package com.example.kotlinassignment

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.Gravity
import android.widget.Button
import android.widget.Toast

class MainActivity2 : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main2)

        val submit = findViewById<Button>(R.id.submit)
        val back = findViewById<Button>(R.id.back)

        submit.setOnClickListener() {
             val linearLayout = null
             val layout = layoutInflater.inflate(R.layout.custom_tag, linearLayout)
             val myToast = Toast(applicationContext)
             myToast.duration = Toast.LENGTH_LONG
             myToast.setGravity(Gravity.CENTER_VERTICAL,0,0)
             myToast.view = layout
             myToast.show()


        }

        back.setOnClickListener() {
            intent = Intent(this, MainActivity::class.java)
            startActivity(intent)
            Toast.makeText(this,"Reverting to 1st Activity", Toast.LENGTH_LONG).show()
        }
    }
}